common/models/user.js

```diff
@@ -306,6 +306,88 @@ module.exports = function(User) {
     }
   );
 
+  User.beforeRemote('updateEmail', function(ctx, result, next) {
+    if (!ctx.req.body.email) {
+      return next();
+    }
+
+    const emailNew = ctx.req.body.email;
+
+    return User.doesExist(null, emailNew)
+      .then(exists => {
+        if (!exists) {
+          return next();
+        }
+
+        ctx.req.flash('error', {
+          msg: dedent`
+      The ${emailNew} email address is already associated with an account.
+      Try using another email address. Or request help at team@freecodecamp.com.
+          `
+        });
+        return ctx.res.redirect('back');
+      })
+      .catch(err => {
+        console.error(err);
+        ctx.req.flash('error', {
+          msg: 'Oops, something went wrong, please try again later'
+        });
+        return ctx.res.redirect('/settings');
+      });
+  });
+
+  User.updateEmail = function updateEmail(email, req, res, cb) {
+    if (!email) {
+      return nextTick(() => {
+        cb(new TypeError(
+            `email should be valid, got ${ email }`
+        ));
+      });
+    }
+    const userId = req.user.id;
+    return User.findById(userId, function(err, user){
+      user.updateAttribute('email',email,function(err){
+        if (err) { return next(err); }
+        debug('updated email for user');
+        req.flash('info', { msg: 'You\'ve successfully updated your email.' });
+        return res.redirect('/settings');
```

Comment:
These details should be handled by the client, so that the API endpoint can be used in different places.